File: WalkingMaps/WalkingMaps/wwwroot/app/core/domain/walk.ts

enum City { Moscow, StPetersburg, Kazan, Sochi }

export class Walk {
    Id: number;
    Name: string;
    Description: string;
    StartPoint: string;
    EndPoint: string;
    CreatedDate: Date;
    NavLink: string;
    Distance: number;
    IsPublished: boolean;
    UserId: string;
    City: City;
    TotalSights: number;
    
    constructor(id: number,
        name: string,
        description: string,
        startPoint: string,
        endPoint: string,
        createdDate: Date,
        navLink: string,
        distance: number,
        isPublished: boolean,
        userId: string,
        city: City,
        totalSights: number) {
        this.Id = id;
        this.Name = name;
        this.Description = description;
        this.StartPoint = startPoint;
        this.EndPoint = endPoint;
        this.CreatedDate = createdDate;
        this.NavLink = navLink;
        this.Distance = distance;
        this.IsPublished = isPublished;
        this.UserId = userId;
        this.City = city;
        this.TotalSights = totalSights;
    }
}



      

File: WalkingMaps/WalkingMaps/Entities/Walk.cs
using System.Collections.Generic;
using System;

namespace WalkingMaps.Entities
{
    public class Walk : IEntityBase
    {
        public Walk()
        {
            WalkSights = new List<WalkSight>();
        }        
       
        public int Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public string StartPoint { get; set; }
        public string EndPoint { get; set; }
        public DateTime CreatedDate { get; set; }
        public string NavLink { get; set; }
        public decimal? Distance { get; set; }
        public bool IsPublished { get; set; }
        public  string UserId { get; set; }
        public City City { get; set; }
        public virtual ICollection<WalkSight> WalkSights { get; set; }
        public virtual Route Route { get; set; }
    }
}
